### ISSUE TITLE
Subtle Trashcan bug when file not found in loop of found files

### ISSUE THREAD

**Issue by ahlstart** [OpenViX/enigma2#796]
In enigma2/lib/python/Tools/Trashcan.py, line 185 in class "CleanTrashTask" method "work" accesses the variable st regardless of whether setting it with "st = os.stat(fn)" on line 182 succeeds or not. The exception (FileNotFoundError) is caught but only a diagnostic is printed, the program logic keeps on as if nothing bad had happened. Adding "continue" on line 184.5 would probably solve this, both for legitimate file disappearances caused by other processes or network drive accessors, and the apparent bug(s) in os module, expanded below.

I bumped into this as follows:
Somehow os.walk() returns a directory named like "xxx.del" among files list, instead of the dir list.
Also the function os.stat fails with that object, raising FileNotFoundError.
Trying to delete that directory just makes it gain another ".del" extension but keeps it on disk.
Someone might want to look at this even though line 184.5 above should solve the crash I encountered.

EDIT: I was careless enough to remove the troubling object and now cannot state anything about its nature - OS module is most probably just fine. Still line 184.5 is missing, in my opinion.

**Comment by ahlstart** [comment]
Revision 011b50 had "pass" on line 185. It was deemed unnecessary. I beg to differ, after experiencing crashes (of the task, not whole Enigma2) caused by having line 185 removed. This caused my trashcan to never permanently remove any files after the troublesome file/folder.

**Comment by TwolDE2** [comment]
Can you post in forum please so we can discuss
Deleting directory should now work and delete a .del(which is added by the c++ code before trying to delete)
I am not at home currently, but probably agree with the continue .. will look when I get back

**Comment by ahlstart** [comment]
Thanks, thread at https://www.world-of-satellite.com/showthread.php?65578-Subtle-Trashcan-bug-when-file-not-found-in-loop-of-found-files

**Comment by TwolDE2** [comment]
Seriously if you want to help and have programming skills and the time, then we would be interested.
  
E2 has been around a long while now, but there are still bugs that need fixing and also we try to add value and functionality where it gives users something…… unlike other images that just rewrite code and add functions that are of no benefit to the users.
  

  
E2 is mostly python with an old C++ core, OpenViX is only python 3.
  

  
The process to build an image takes some understanding, but in reality is not that difficult if you have an IT background.
  

  
Main guy is Huevos (lives in Spain) and is really good at python, I live in Germany.
  
  
  
  
  
  
  
Best Regards
  
Tony
  
  
>

**Comment by ahlstart** [comment]
Yep, I have been thinking about contributing for a few years but was taken aback by the apparent complexity of building images.
I am almost certain it is actually straightforward and logical once I see it done once. Or twice.

OpenViX was my first image on my Vu+Uno4KSE, I tried others when bugs hit my usage patterns but am solidly back with version 6.2; I have very few issues left. Which will be fixed at low priority.

Feel free to close this issue once you are happy with the code, we have email & forum to communicate about contributing.

Best Regards,
Arto

**Comment by Huevos** [comment]
Building is pretty simple. Just follow the readme at the root of this repo. And if you have any problems we can help on the forum. People are even building on Windows inside a virtual machine, but preferably you require a dedicated pc or server machine with Ubuntu Linux operating system.

**Comment by ahlstart** [comment]
Yes, it looked simple enough. Until I arrived at site.conf file tweaking. No such file - until I tried building anyways. Seems that it was built by the makefile just fine.

 

My install on Windows Ubuntu 20.4.5 "app" did not go over well, probably because I could not do a chown to moved-to-bigger-drive home directory of the new user openvixbuilder so that it would have stuck. So bitbakery went up in flames. Will try again on a virtual machine with enough space on home directory drive (bitlocker makes multi-booting a frightening proposition).
